### PR TITLE
[NetApp] netapp microsoft.net app 2023 05 01

### PR DESCRIPTION
**Pull request by audunn**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by audunn** [comment]
Breaking changes previously reviewed and approved in https://github.com/Azure/azure-rest-api-specs-pr/pull/8331

**Comment by ms-henglu** [comment]
Please check and fix the Avocado errors.

**Comment by audunn** [comment]
> Please check and fix the Avocado errors.
@ms-henglu Those errors are related to the approved breaking changes

**Comment by konrad-jamrozik** [comment]
@audunn the Avocado check is no longer required to merge a PR.  You can confirm this by observing the check `Automated merging requirements met` no longer mentions Avocado and by lack of the `Required` label beside Avocado in the list of failing checks at the bottom of the PR. Please proceed as explained in the `Next Steps to Merge` comment and the PR description diagram. Thanks!

**Comment by sjanamma** [comment]
>     "/subscriptions/{subscriptionId}/providers/Microsoft.NetApp/locations/{location}/queryNetworkSiblingSet": {

Can these two POST calls be modelled as GET and PUT calls?

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2023-05-01/netapp.json:332 in 4569115. [](commit_id = 45691155e65c67e45eda841b975aab68077d0bf5, deletion_comment = False)

**Comment by sjanamma** [comment]
>       "title": "Subnet resource Id",

consider formatting the property as an arm-id. [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details)

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2023-05-01/netapp.json:7388 in 4569115. [](commit_id = 45691155e65c67e45eda841b975aab68077d0bf5, deletion_comment = False)

**Comment by audunn** [comment]
> > ```
> > "/subscriptions/{subscriptionId}/providers/Microsoft.NetApp/locations/{location}/queryNetworkSiblingSet": {
> > ```
> 
> Can these two POST calls be modelled as GET and PUT calls?
> 
> Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2023-05-01/netapp.json:332 in [4569115](https://github.com/Azure/azure-rest-api-specs/commit/45691155e65c67e45eda841b975aab68077d0bf5). [](commit_id = [4569115](https://github.com/Azure/azure-rest-api-specs/commit/45691155e65c67e45eda841b975aab68077d0bf5), deletion_comment = False)

Not realy this is not creating any resoruces, it is querying the backend for specific data this cannot be done within the GET latency requrements.

**Comment by mentat9** [comment]
>     "volumeResourceIds": {

Looks like this could also benefit from formatting as arm-id for stronger typing and better consistency.

---
Refers to: specification/netapp/resource-manager/Microsoft.NetApp/stable/2023-05-01/netapp.json:7379 in 03c360f. [](commit_id = 03c360fb5232c6f48cbb239b0e7d6b71e7b83cbe, deletion_comment = False)

**Comment by mentat9** [comment]
Signed off for ARM with comments. Please take a look and consider making the suggested change before merge.

**Comment by audunn** [comment]
@mentat9 out of curiosity if we were to add arm-id to an existing property for consitency sake,  would that be a breaking change, do you know?

**Comment by mentat9** [comment]
> @mentat9 , @sjanamma out of curiosity if we were to add arm-id to an existing property for consitency sake, would that be a breaking change, do you know?

@audunn - Looks like the automation answered your question by adding the BreakingChangeReviewRequired label. Note I didn't require that change for ARM signoff, but to merge, you will need to either revert it or get breaking change approval. I imagine it should pass breaking change review easily since it doesn't represent a change from your backend (IIUC).

**Comment by audunn** [comment]
> arm-id

@mentat9 That seems strange, perhaps a false positive here, as we added arm-id to a new property introduced in this api-version not to an existing one.  We need to look into why it flags

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2023-05-01/netapp.json
Comment (mikekistler) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2023-05-01/netapp.json:7384]: Did you want to add `"format": "arm-id",` here?

**Comment by audunn** [comment]
Prettier fails , does not fail locally something with the pipeline 

```
prettier -c specification/netapp/resource-manager/Microsoft.NetApp/stable/2023-05-01/examples/Volumes_Update.json
[error] Cannot find module 'prettier/plugins/estree'
[error] Require stack:
[error] - /mnt/vss/_work/1/azure-rest-api-specs/eng/scripts/prettier-swagger-plugin.js
[error] - /mnt/vss/_work/1/a/unified-pipeline-runtime/common/temp/node_modules/.pnpm/prettier@2.1.2/node_modules/prettier/index.js
[error] - /mnt/vss/_work/1/a/unified-pipeline-runtime/common/temp/node_modules/.pnpm/prettier@2.1.2/node_modules/prettier/bin-prettier.js

```

**Review by mentat9 — APPROVED** [review]

**Pull request merged**
